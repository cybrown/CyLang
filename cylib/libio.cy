void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
