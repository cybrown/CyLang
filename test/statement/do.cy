import libio;

int a;

a = 0;
do
{
	putint(a);
	putchar(10);
	a += 1;
} while (a < 10);

putchar(10);

a = 20;
do
{
	putint(a);
	putchar(10);
	a -= 1;
} while (a > 0);

putchar(10);

a = 100;
do
{
	putint(a);
	putchar(10);
	a -= 5;
} while (a > -20);

putchar(10);
