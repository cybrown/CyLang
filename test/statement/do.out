0
1
2
3
4
5
6
7
8
9

20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1

100
95
90
85
80
75
70
65
60
55
50
45
40
35
30
25
20
15
10
5
0
-5
-10
-15

